File: 1557. Minimum Number of Vertices to Reach All Nodes - LeetCode Problem/MinimumNumberofVerticestoReachAllNodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        int[] inDegree = new int[n];
        
        for (List<Integer> edge : edges)
            inDegree[edge.get(1)]++;
        
        
        List<Integer> ans = new ArrayList<>();
        for (int i = 0; i < n; i++)
            if (inDegree[i] == 0)
                ans.add(i);
        
        return ans;
    }
}

File: SquidGames/Round Four/Feb 21 Daily Questions/1557. Minimum Number of Vertices to Reach All Nodes/MinimumNumberofVerticestoReachAllNodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        indegree = [0 for _ in range(n)]
        for _from, _to in edges:
            indegree[_to] += 1
        
        return [i for i, x in enumerate(indegree) if not x]